weatherman.py

```diff
@@ -0,0 +1,167 @@
+                    daily_data = daily_data.split(',')
+                    year = daily_data[0].split('-')[0]
+                    date = daily_data[0]
+                    max_temp = int_parse_wrapper(daily_data[1])
+                    min_temp = int_parse_wrapper(daily_data[3])
+                    max_humid = int_parse_wrapper(daily_data[7])
+                    min_humid = int_parse_wrapper(daily_data[9])
+
+                    each_year = self.year_data.get(year)
+
+                    if not each_year:
+                        self.year_data[year] = WeatherYearData()
+                        each_year = self.year_data[year]
+
+                    each_year.get_data(date, max_temp, min_temp, max_humid,
+                                       min_humid)
+
+    def print_report(self, report_number):
+        """Used for printing yearly reports.
+        User gets 2 options.
+        1 for yearly max/min temperature and max/min humidity
+        2 for hottest day of the year and the temperature on that day.
+        """
+
+        if report_number == 1:
+            print('{:18}{:14}{:16}{:16}{}'.format('Year', 'MAX Temp',
+                                                  'MIN Temp', 'MAX Humidity',
+                                                  'MIN Humidity'))
+            for key, each_year_data in sorted(self.year_data.items()):
+                print(f'{key} {check_data_report(each_year_data.max_temp, DEFAULT_MAX):15}'
+                      f'{check_data_report(each_year_data.min_temp, DEFAULT_MIN):15}'
+                      f'{check_data_report(each_year_data.max_humid, DEFAULT_MIN):15}'
+                      f'{check_data_report(each_year_data.min_humid, DEFAULT_MIN):15}')
+        elif report_number == 2:
+            print('{:18}{:15}{}'.format('Year', 'MAX Temp', 'DATE'))
+            for key, each_year_data in sorted(self.year_data.items()):
+                print(f'{key} {check_data_report(each_year_data.max_temp, DEFAULT_MAX):15}             '
+                      f'{check_data_report(each_year_data.max_temp_date, DEFAULT_STR):15}')
+
+
+def int_parse_wrapper(data):
+    """Converts data to integer format."""
+    try:
+        return int(data)
+    except ValueError:
+        return None
+
+
+def check_data_report(value, default):
+    """Checks if the value is default. Returns '-' if true, or returns the
+    original value.
+    """
+
+    if value == default:
+        return '-'
+    else:
+        return value
+
+
+def check_directory(data_dir):
+    """Argparser function, created to check type for the data_dir argument.
+    Checks if the path exists and  confirms if the direcory isn't empty.
+    If all the conditions are satisfied, it accepts the argument.
+    """
+
+    if os.path.isdir(data_dir):
+        file_list = os.listdir(data_dir)
+        if not len(file_list) == 0:
+            return data_dir
+        else:
+            msg = '%r is an empty directory' % data_dir
```

Comment:
try to use same type of string formating to maintain consistency.
Could use f strings used above while printing.